consensus/spos/consensusMessageValidator.go

```diff
@@ -0,0 +1,270 @@
+package spos
+
+import (
+	"bytes"
+	"fmt"
+
+	"github.com/ElrondNetwork/elrond-go-logger"
+	"github.com/ElrondNetwork/elrond-go/consensus"
+	"github.com/ElrondNetwork/elrond-go/core"
+)
+
+func (wrk *Worker) checkConsensusMessageValidity(cnsMsg *consensus.Message) error {
+	if !bytes.Equal(cnsMsg.ChainID, wrk.chainID) {
+		return fmt.Errorf("%w : received chain ID from consensus topic is invalid: %s",
+			ErrInvalidChainID,
+			logger.DisplayByteSlice(cnsMsg.ChainID))
+	}
+
+	err := wrk.checkConsensusMessageValidityForMessageType(cnsMsg)
+	if err != nil {
+		return err
+	}
+
+	msgType := consensus.MessageType(cnsMsg.MsgType)
+
+	isBlockHeaderHashSizeInvalid := wrk.consensusService.IsMessageWithBlockBody(msgType) && cnsMsg.BlockHeaderHash != nil ||
```

Comment:
there is a need for paranthesis here - it does not work as expected. on set till || and the other set after ||

Reply:
Refactored

Reply:
funny fact in GO - || has precedence over &&